theano/gof/type.py

```diff
@@ -606,8 +606,8 @@ def __str__(self):
 
 generic = Generic()
 
-_cdata_type = ctypes.py_object.from_address(
-    ctypes.addressof(ctypes.pythonapi.PyCapsule_Type)).value
+#_cdata_type = ctypes.py_object.from_address(
```

Comment:
What about:
_cdata_type = None
if not pypy:
   _cdata_type = ...

This would remove the check, but should make it work.